### ISSUE TITLE
App migration from emulated sdcard internal to data internal fails

### ISSUE THREAD

**Issue by xpaixx** [vishnunaini/lineage-surnia-releases#8]
Basically the topic describe the issue. Apps are installed on SD card (formatted as internal storage) by default and when trying to move them to internal storage (phone) I receive a communicate that there's not enough space, even when trying to move <10 MB apps.

**Comment by vishnunaini** [comment]
This is a known bug that also exists upstream. I wan't to fix bugs but I can only fix these complicated bugs after long investigations, bisects and lot of rebuilds. I will look into it if I have time.

**Comment by vishnunaini** [comment]
Since workarounds are possible I will consider this a low priority.

Does the workaround of moving all data from sdcard to internal work for you? Use the migrate data option in the internal storage menu.

**Comment by xpaixx** [comment]
Migrate is an option when moving apps FROM phone TO sd and can't be done in the opposite way. I know this bug was present in latest CM 14 build, but this way I'll know when it's done.

**Comment by vishnunaini** [comment]
Check in the 3 dot menu in the phone section and click on migrate data. You
will also get an option to move to phone from SD.

**Comment by xpaixx** [comment]
Well I've tried ejecting the SD card and then installing apps.
>card is being disconnected
>phone reboots
>back in the system, card is still on
>ejected it one more time
>apps still fail to install
>reconnects the card
>card is formatted with only _Android_ folder on it

The card was and still is encrypted, but EVERY data except for _Android_ is gone. I think You can consider is as a different bug.

**Comment by vishnunaini** [comment]
You shouldn't eject adopted storage sdcard while system is running. Android will consider apps as uninstalled and wipe userdata.

**Comment by xpaixx** [comment]
> When you eject this SD card, apps stored on it will stop working, and media files stored on it will not be available until it is reinstalled.

Being unavailable and being wiped are two different things. Next time I've done it nothing was wiped. Same on an other ROM.

**Comment by vishnunaini** [comment]
Just wait on upstream fixing this bug. I wanted to fix it on my own but can't make time to bisect the bug as the logs are of not much help. Bisecting requires a lot of rebuilds.
